edge/key.go

```diff
@@ -20,15 +21,26 @@ type edgeKey struct {
 
 // SetKey parses and associates an Edge key to the agent.
 // If the agent is running inside a Swarm cluster, it will also set the "set" flag to specify that a key is set on this agent in the cluster.
+// Don't overwrite the file if it exists - this only appears to work, because our container runs as root
```

Comment:
maybe as non container, we should expect agent.DataDirectory to be writeable by this procss?

Reply:
set key should run once, we don't give an option to replace keys. so I guess:
1. first get from env-vars/flags.
2. if doesn't exist, try to get from fs
3. if doesn't exist, try to get from cluster
4. if doesn't exist, try to get from edge ui
5. fail after X minutes
6. save and rewrite once you get the key

Reply:
yup - i kinda think this kind of change should be speced out, to make sure what is implemented is known and understood.

Reply:
I left a comment in the other PR, I'm replicating it here just in case: https://github.com/portainer/agent/pull/231/files#r799889829